test-case/check-capture.sh

```diff
@@ -65,21 +65,21 @@ func_pipeline_export "$tplg" "type:capture & ${OPT_VAL['S']}"
 
 for round in $(seq 1 $round_cnt)
 do
-    for idx in $(seq 0 $(expr $PIPELINE_COUNT - 1))
+    for idx in $(seq 0 $((PIPELINE_COUNT - 1)))
     do
-        channel=$(func_pipeline_parse_value $idx channel)
-        rate=$(func_pipeline_parse_value $idx rate)
-        fmt=$(func_pipeline_parse_value $idx fmt)
-        dev=$(func_pipeline_parse_value $idx dev)
-        pcm=$(func_pipeline_parse_value $idx pcm)
-        type=$(func_pipeline_parse_value $idx type)
-        snd=$(func_pipeline_parse_value $idx snd)
+        channel=$(func_pipeline_parse_value "$idx" channel)
+        rate=$(func_pipeline_parse_value "$idx" rate)
+        fmt=$(func_pipeline_parse_value "$idx" fmt)
+        dev=$(func_pipeline_parse_value "$idx" dev)
+        pcm=$(func_pipeline_parse_value "$idx" pcm)
+        type=$(func_pipeline_parse_value "$idx" type)
+        snd=$(func_pipeline_parse_value "$idx" snd)
 
         if [ ${OPT_VAL['F']} = '1' ]; then
-            fmt=$(func_pipeline_parse_value $idx fmts)
+            fmt=$(func_pipeline_parse_value "$idx" fmts)
         fi
 
-        for fmt_elem in $(echo $fmt)
+        for fmt_elem in $fmt
```

Comment:
The previous code was really weird, I don't understand what it was trying to achieve so I'm a bit concerned that the fix seems "too simple". @aiChaoSONG you added this code in commit aedc851c46d0, can you please review this PR?

Reply:
@greg-intel can you please test the -F option before and after this PR and compare the logs, try to spot any difference?

Reply:
I didn't spot a difference, but I think maybe the device only had one format.  I *did* (during testing) check behavior of this line when there were two items in the list, and it iterated over each individually.  (I added " testing" to fmt, it ran the known value first and the second generated an error because it didn't exist.)

Reply:
I could not recall why it is `$(echo $fmt)`, let's do a test see if `$fmt` is ok, if yes, it will be good then.